tests/test_dataclasses.py

```diff
@@ -301,3 +301,28 @@ class User:
         },
         'required': ['id'],
     }
+
+
+def test_nested_schema():
+    @pydantic.dataclasses.dataclass
+    class Nested:
+        number: int
+
+    @pydantic.dataclasses.dataclass
+    class Outer:
+        n: Nested
+
+    assert Outer.__pydantic_model__.schema() == {
```

Comment:
maybe we should add a method to pydantic dataclasses `schema()`? I know it could conflict with a field, but I suspect things break if you call a field `schema` already.

Reply:
In the case of adding `schema` method, do we also want to add `schema_json`, `json`, `parse_file`, and other methods?

From a user's pespective, I think it would be nice to have unified interfaces among dataclass-based models and BaseModel models. But it might be out of scope for this issue. I guess it might be better to open a new issue to discuss this?

Reply:
Humm, I get the idea, but I also want pydantic dataclasses to be as close to vanilla as possible.

For now let's leave it as just `Outer.__pydantic_model__.schema()` ugly, but works.

Reply:
#481, for another PR.